backend/dal/Services/Admin/Concrete/UserService.cs

```diff
@@ -141,88 +151,109 @@ public User Get(Guid id)
-            var removeRoles = user.Roles.Except(entity.Roles, new UserRoleRoleIdComparer());
+            var removeRoles = existingUser.Roles.Except(user.Roles, new UserRoleRoleIdComparer());
             removeRoles.ForEach(r => this.Context.Entry(r).State = EntityState.Deleted);
 
-            var addAgencies = entity.Agencies.Except(user.Agencies, new UserAgencyAgencyIdComparer());
+            var addAgencies = user.Agencies.Except(existingUser.Agencies, new UserAgencyAgencyIdComparer());
             addAgencies.ForEach(a => this.Context.Entry(a).State = EntityState.Added);
-            var removeAgencies = user.Agencies.Except(entity.Agencies, new UserAgencyAgencyIdComparer());
+            var removeAgencies = existingUser.Agencies.Except(user.Agencies, new UserAgencyAgencyIdComparer());
             removeAgencies.ForEach(a => this.Context.Entry(a).State = EntityState.Deleted);
 
-            base.Update(entity);
-            this.Context.Detach(entity);
+            base.Update(user);
+            this.Context.Detach(user);
         }
 
         /// <summary>
         /// Remove the specified user from the datasource.
         /// </summary>
         /// <exception type="KeyNotFoundException">Entity does not exist in the datasource.</exception>
-        /// <param name="entity"></param>
-        public override void Remove(User entity)
+        /// <param name="user"></param>
+        public override void Remove(User user)
         {
-            entity.ThrowIfNull(nameof(entity));
+            user.ThrowIfNull(nameof(user));
 
-            var user = this.Context.Users
+            var existingUser = this.Context.Users
                 .Include(u => u.Agencies)
                 .Include(u => u.Roles)
                 .AsNoTracking()
-                .FirstOrDefault(u => u.Id == entity.Id) ?? throw new KeyNotFoundException();
+                .FirstOrDefault(u => u.Id == user.Id) ?? throw new KeyNotFoundException();
 
-            this.Context.SetOriginalRowVersion(user);
+            this.Context.SetOriginalRowVersion(existingUser);
 
-            user.Roles.Clear();
-            user.Agencies.Clear();
+            existingUser.Roles.Clear();
+            existingUser.Agencies.Clear();
 
-            base.Remove(user);
+            base.Remove(existingUser);
         }
 
         /// <summary>
         /// Update the database using the passed AccessRequest
         /// </summary>
-        /// <param name="entity"></param>
-        public AccessRequest UpdateAccessRequest(AccessRequest entity)
+        /// <param name="accessRequest"></param>
+        public AccessRequest UpdateAccessRequest(AccessRequest accessRequest)
         {
-            var accessRequest = GetAccessRequest(entity.Id);
-            entity.UpdatedById = this.User.GetUserId(); // TODO: No longer needed.
-            entity.UpdatedOn = DateTime.UtcNow;
-            this.Context.Entry(accessRequest).CurrentValues.SetValues(entity);
```

Comment:
This is dangerous and causes more problems than it is worth.  It is almost always better to manually identify what needs to be updated.